allennlp/data/vocabulary.py

```diff
@@ -9,26 +9,37 @@
 
 class _NamespaceDependentDefaultDict(defaultdict):
     """
-    Using ``defaultdicts`` to handle namespaces is really convenient - you don't need to worry
-    about whether you've seen a particular namespace before in any of the methods in
-    ``Vocabulary``.  But because some namespaces need padding (like "tokens") and some don't (like
-    "labels"), we want different defaults depending on the namespace.  This lets us still use a
-    ``defaultdict``, but have different default values depending on the key that we're given.
+    Sometimes certain namespaces need padding (like "tokens") and some don't (like
+    "labels"), and we want different defaults depending on the namespace.  This class lets us use a
+    ``defaultdict`` (https://docs.python.org/2/library/collections.html#collections.defaultdict),
+    but have different default values depending on the namespace of the key.
+
+    This class also handles *-namespaces.  In other words, if "*tags" is in non_padded_namespaces
+    then "passage_tags", "question_tags", etc. (anything that ends with "tags" will have the
+    non_padded default value.
     """
     def __init__(self, non_padded_namespaces: List[str], padded_function, non_padded_function):
         self._non_padded_namespaces = non_padded_namespaces
         self._padded_function = padded_function
         self._non_padded_function = non_padded_function
         super(_NamespaceDependentDefaultDict, self).__init__()
 
+    """
+    Matches a namespace pattern against a namespace string.  For example, "*tags" matches
+    "passage_tags" and "question_tags" and "tokens" matches "tokens" but not "stemmed_tokens".
+    """
+    def _namespace_match(self, pattern: str, namespace: str):
```

Comment:
Is there somewhere central I can put this method?  I'd love to re-use it rather than have the `namespace.endswith(pattern[1:])` logic over and over again--it's not terribly complex but it scatters the documentation.